packages/slate-react/src/components/leaf.js

```diff
@@ -125,9 +125,11 @@ class Leaf extends React.Component {
   renderText() {
     const { block, node, parent, text, index, leaves } = this.props
 
-    // COMPAT: If the text is empty and it's the only child, we need to render a
-    // <br/> to get the block to have the proper height.
-    if (text == '' && parent.kind == 'block' && parent.text == '') return <br />
+    // COMPAT: If the text is empty and it's the only child, we need to render
+    // an invisible unicode char to get the block to have the proper height.
+    if (text == '' && parent.kind == 'block' && parent.text == '') {
+      return '\u2060'
```

Comment:
I believe there are other places in the code base that we were checking for the case where we rendered a `<br />` as the only child. Could you double-check that changing this won't break those assumptions? Or update them if so.

Also how did you choose `'\u2060'` btw?

Reply:
Firstly, I search for `<br />` in the code base and the only match is the code in `slate-html-serializer`, whose `<br />` is used merely to render string markup. Neither does deserializing string with `'br'` affects editor's behavior itself. So after checking I consider it save.

Secondly, `'\u2060'` [word-joiner](https://en.wikipedia.org/wiki/Word_joiner) is chosen because it prevents line break on its either side, and it's invisible, so its semantic seems more appropriate than `'\u200B'`zero width space. Smoke tests are run on Firefox, Chrome and Safari.